src/coord/src/coord.rs

```diff
@@ -1749,36 +1985,55 @@ where
                 SequencedCommand::DropSources(sources_to_drop),
             );
         }
-        if !views_to_drop.is_empty() {
-            for id in views_to_drop {
-                self.views.remove(&id);
-            }
-        }
         if !sinks_to_drop.is_empty() {
             broadcast(
                 &mut self.broadcast_tx,
                 SequencedCommand::DropSinks(sinks_to_drop),
             );
         }
         if !indexes_to_drop.is_empty() {
-            // It might be a table; just blindly remove it from `self.tables`
-            // in case it is.
-            for (id, _) in &indexes_to_drop {
-                let _ = self.tables.remove(id);
-            }
             self.drop_indexes(indexes_to_drop);
         }
 
         Ok(())
     }
 
-    fn import_source_or_view(&self, id: &GlobalId, dataflow: &mut DataflowDesc) {
+    fn drop_sinks(&mut self, dataflow_names: Vec<GlobalId>) {
+        broadcast(
+            &mut self.broadcast_tx,
+            SequencedCommand::DropSinks(dataflow_names),
+        )
+    }
+
+    fn drop_indexes(&mut self, indexes: Vec<GlobalId>) {
+        let mut trace_keys = Vec::new();
+        for id in indexes {
+            if self.indexes.remove(&id).is_some() {
+                trace_keys.push(id);
+            }
+        }
+        if !trace_keys.is_empty() {
+            broadcast(
+                &mut self.broadcast_tx,
+                SequencedCommand::DropIndexes(trace_keys),
+            )
+        }
+    }
+
+    /// Imports the view, source, or table with `id` into the provided
+    /// dataflow description.
+    fn import_into_dataflow(&self, id: &GlobalId, dataflow: &mut DataflowDesc) {
         if dataflow.objects_to_build.iter().any(|bd| &bd.id == id)
             || dataflow.source_imports.iter().any(|(i, _)| i == id)
         {
             return;
         }
-        if let Some((index_id, keys)) = &self.views[id].default_idx {
+        // A valid index is any index on `id` that is known to the dataflow
+        // layer, as indicated by its presence in `self.indexes`.s
```

Comment:
Nit: extra s at end of sentence

Reply:
Thanks, will fix in a follow up to avoid going through CI again.